### ISSUE TITLE
Add an option to force media to play without a letterbox

### ISSUE THREAD

**Issue by anarion80** [dermotduffy/frigate-hass-card#755]
**Checklist:**

- [X] I updated to the latest version available
- [X] I cleared the cache of my browser

**Release with the issue:**
v3.0.0.

**Last working release (if known):**
N/A

**Browser and Operating System:**
Windows 10, Firefox 103.0b4 and Chrome 103.0.5060.66

**Description of problem:**

When going full screen, the camera feed is stretched to the maximum, cutting much of the feed:
![frigate_card_fullscreen1](https://user-images.githubusercontent.com/2185791/177396167-f5347755-caf2-4e69-bfe4-6dc113b82b0b.png)

*I would rather expect that it fits height instead of width. Is this per design or is there some config option I need to change?*

**Additional information:**
My config:
```
 - type: custom:frigate-card
   cameras:
     - camera_entity: camera.main_2
     - camera_entity: camera.garage_2
     - camera_entity: camera.garden_2
   view:
     default: live
     timeout_seconds: 0
     update_cycle_camera: true
     actions:
       double_tap_action:
         action: custom:frigate-card-action
         frigate_card_action: fullscreen
   dimensions:
     aspect_ratio_mode: dynamic
     #aspect_ratio: '1/1'
   card_mod:
     style:
       .: |
         ha-card {
           border-radius: 4.75% !important;
         }
         .container {
           border-radius: 4.75% !important;
         }
```

The stream is 2560x1920, screenshot from ultrawide monitor.

**Comment by dermotduffy** [comment]
Hi @anarion80, the expected behavior is that it fills the screen without losing any content by letterboxing the content. I can't tell from the screenshot, but presumably that is not happening here and you're losing content off the bottom (so I agree this sounds like a bug).

As a quick try, can you use the latest build just to confirm whether or not you see the same beahvior in your usecase: https://github.com/dermotduffy/frigate-hass-card/releases/tag/v4.0.0-rc.2

**Comment by anarion80** [comment]
Thanks for the reply. Yes, the problem was that feed was cut off at the bottom.
After switching to v4.0.0.-rc.2 the situation is fixed:
![frigate_card_fullscreen3](https://user-images.githubusercontent.com/2185791/177489611-9b8a2e75-cf0e-40e4-8dd1-ce2065913fb4.png)

Thanks for support!
I will close the issue, but I have one more possibly related question. Is there a way to make the card fill entire view (so fill height)?
I have square 'buttons' that I want to show the feed.

So how to make this:
![obraz](https://user-images.githubusercontent.com/2185791/177493957-d3e92a0e-9a3e-4319-8e01-7d099ff3ac5f.png)

into this:
![obraz](https://user-images.githubusercontent.com/2185791/177494002-a546158a-2ad0-493f-bf08-01e7f5402a82.png)


?

**Comment by dermotduffy** [comment]
@anarion80 Just to clarify, I think your request is:

* Could the camera live view *intentionally* be clipped (losing content) in order to fill both dimensions (width & height) without letterboxing (the black strips). 

If that's the request, this is not possible today.

**Comment by anarion80** [comment]
Yes, it's the first one - intentionally clip on both sides just to fill both dimensions.

**Comment by dermotduffy** [comment]
@anarion80 Do you imagine this is something you'd like to set differently per camera, or once for the whole card?

**Comment by anarion80** [comment]
In my use case - it would be per card.

**Comment by conorlap** [comment]
>
>     * Could the camera live view _intentionally_ be clipped (losing content) in order to fill both dimensions (width & height) without letterboxing (the black strips).
> 

I was about to request something similar to this! My request is the ability to 'crop in' on the live video feed coupled with the ability to utilize the aspect ratio changer on the card to allow a "full mobile screen" of my front door.

My regular front door feed with WebRTC:
<img width="418" alt="Screenshot 2022-08-01 at 16 22 30" src="https://user-images.githubusercontent.com/57415184/182183604-c0da4216-b74f-4c81-b30a-ca5df5dd832f.png">

The cropped feed that would take up more of the screen on my mobile app:
<img width="420" alt="Screenshot 2022-08-01 at 16 22 42" src="https://user-images.githubusercontent.com/57415184/182183612-8c1f7474-29fa-487a-8b77-414b1284e29b.png">

I was thinking in terms of use case, I'd have this 'cropped stream dashboard' appear when tapping a motion detection or doorbell press notification. It would make the HA experience more akin to the Nest Doorbell. 

Today I've achieved a cropped feed by using the camera proxy built into HA, but this is a very low quality feed, isn't WebRTC and doesn't have audio etc.

```
camera:
  - platform: proxy
    entity_id: camera.amcrest_camera
    name: My cropped camera
    mode: crop
    image_top: 0
    image_left: 230
    max_image_width: 360
    max_image_height: 950
    max_stream_width: 360
    max_stream_height: 1000
    stream_quality: 100
    force_resize: true
```

**Comment by dermotduffy** [comment]
Still planning to experiment with this!

I think we can do some of this by allowing the user to control the [object-fit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) and [object-position](https://developer.mozilla.org/en-US/docs/Web/CSS/object-position) of the various forms of media the card supports. Notably absent is "zoom" / scale, I can't find a clever way to do that that doesn't result in the whole video component being scaled (i.e. including the video controls, possibly cutting them off, etc).

In short: I think this would cover @anarion80's usecase, and at least some of @conorlap's ...

**Comment by dermotduffy** [comment]
This is now implemented, see https://github.com/dermotduffy/frigate-hass-card/tree/release-4.0.0#changing-the-media-layout for examples of usage. You'll need to build your own version to test it out, or wait for the next official build.

**Comment by conorlap** [comment]
Working a treat here @dermotduffy - thank you!! 🎉

**Comment by anarion80** [comment]
I finally got around to using this further, and it works here as well. Big thanks @dermotduffy !

![obraz](https://user-images.githubusercontent.com/2185791/190866368-e752e6af-e929-4e68-94ef-c9e940337633.png)

However, I'm probably missing some config value, because now the button looks stretched as it should, but when double clicking on it, the full screen view is also streched to fit width, so now cutting a lot...
How do I disassociate the setting for the card and its full screen view?

**Comment by dermotduffy** [comment]
@anarion80 I think that's working as intended, but I think you're asking (essentially) for a different configuration depending on whether you're in fullscreen or not. Am I understanding correctly? If yes, you might be able to override the config -- mind sharing your configuration and I'll take a look?

**Comment by anarion80** [comment]
Yes, here's the card config:

```
- type: custom:frigate-card
  cameras:
    - camera_entity: camera.main_2
      live_provider: ha
    - camera_entity: camera.garage_2
      live_provider: ha
    - camera_entity: camera.garden_2
    - camera_entity: camera.main_2
      live_provider: webrtc-card
      title: Front Door (WebRTC)
      webrtc_card:
        url: 'rtsp://user:pass@192.168.2.48:554/h264Preview_01_main'
      id: front-door-webrtc
  view:
    # scan:
    #   enabled: false
    default: live
    timeout_seconds: 300
    update_seconds: 5
    update_force: false
    actions:
      double_tap_action:
        action: custom:frigate-card-action
        frigate_card_action: fullscreen
  live:
    layout:
      fit: cover
    controls:
      thumbnails:
        mode: none
      title:
        mode: none
```
For the fullscreen I would like it to be exactly like in 3rd post in this thread ;)

**Comment by dermotduffy** [comment]
@anarion80 This kind of thing should work, play around with `overrides` as shown below:

```
type: custom:frigate-card
cameras:
  - camera_entity: camera.landing
  - camera_entity: camera.living_room
  - camera_entity: camera.sitting_room
view:
  default: live
  timeout_seconds: 300
  update_seconds: 5
  update_force: false
  actions:
    double_tap_action:
      action: custom:frigate-card-action
      frigate_card_action: fullscreen
live:
  layout:
    fit: cover
  controls:
    thumbnails:
      mode: none
    title:
      mode: none
overrides:
  - conditions:
      fullscreen: true
    overrides:
      live:
        layout:
          fit: contain
```

Give it a shot. Note that I found a bug on coming out of fullscreen in this configuration, it doesn't reset correctly back (filed: https://github.com/dermotduffy/frigate-hass-card/issues/830 ). You may or may not run into the same issue.

**Comment by anarion80** [comment]
And it does work indeed! Thank you very much!
